src/NuGetPackageConfigConverter/Services/PackageManagerConverter.cs

```diff
@@ -183,57 +247,113 @@ private static void RemoveDependencyFiles(Project project)
             project.Save();
         }
 
-        private static void RefreshSolution(Solution sln, IEnumerable<Project> projects)
+        private void RefreshSolution(Solution sln, IEnumerable<Project> projects, ConverterUpdateViewModel model)
         {
-            var projectPaths = projects.Select(p => p.FullName).ToList();
-            var path = sln.FullName;
+            try
+            {
 
-            sln.Close();
 
-            foreach (var project in projectPaths)
+                var projectInfos = projects.Select(p => new ProjectInfo(p.GetFullName(), p.Name)).ToList();
+                var slnPath = sln.FullName;
+                logger.Log("Closing solution");
+                sln.Close();
+                int counter = 0;
+                int total = projectInfos.Count(p => !string.IsNullOrEmpty(p.FullName));
+                foreach (var project in projectInfos.Where(p => !string.IsNullOrEmpty(p.FullName)))
+                {
+                    counter++;
+                    model.Status = $"Fixing restore style in'{project.Name}/{project.FullName}'   ({counter}/{total}";
+                    logger.Log(model.Status);
+                    AddRestoreProjectStyle(project.FullName, project, model);
+                }
+
+                sln.Open(slnPath);
+                logger.Log("Reopen the solution");
+            }
+            catch (Exception e)
             {
-                AddRestoreProjectStyle(project);
+                model.Log = $"Exception while working with restore style property.  Do this manually.";
+                logger.Log(model.Log);
+                logger.Log(e.ToString());
+
             }
 
-            sln.Open(path);
+
         }
 
-        private static void AddRestoreProjectStyle(string path)
+
+        class ProjectInfo
```

Comment:
Can you move the inner classes to their own files?

Reply:
yep